stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -302,16 +304,23 @@ const sorted_keyvals = ["false", "true"]
 complete_keyval!(suggestions::Vector{Completion}, s::String) =
     complete_from_list!(suggestions, KeyvalCompletion, sorted_keyvals, s)
 
-function do_raw_escape(s)
-    # escape_raw_string with delim='`' and ignoring the rule for the ending \
-    return replace(s, r"(\\+)`" => s"\1\\`")
+function do_cmd_escape(s)
+    return Base.shell_escape_posixly(Base.escape_raw_string(s, '`'))
```

Comment:
This seems like nonsense: I can't think of any case where we'd expect it'd be useful for the the REPL to pass the string unmodified to a posix shell for use as an argument in a julia script, which is what this transform order implements.

The previous transform took a string `text` that was already in posix-shell form (e.g. for raw input to shell_parse for `shell> text` mode) and corrects it for the julia parser (e.g. for ``:(`text`)``)
```
julia> a = `\\ b\\\\`
`'\' 'b\'`
```

Reply:
I intended `cmd_escape` to make it possible to complete shell commands inside Cmd-strings, but mixed up the order (``Base.escape_raw_string(Base.shell_escape_posixly(s), '`')`` does what I want I think):
```
julia> println(R.do_cmd_escape("file ` 1"))
'file \` 1'

julia> @macroexpand `'file \` 1'`
:(Base.cmd_gen((("file ` 1",),)))
```

Reply:
Completion with `shell_escape_posixly` is still unwieldy because you get partial completions that insert only the opening quote, and can't complete any further.